File: abc/abc140/abc140_c.cs
// https://atcoder.jp/contests/abc140/submissions/7390650

using System;
using System.Linq;

class Program
{
    static void Main()
    {
        var N = int.Parse(Console.ReadLine());
        var B = Console.ReadLine().Split(' ').Select(int.Parse).ToArray();

        var A = new int[N];

        A[0] = B[0];
        for (int i = 1; i < N - 1; i++)
        {
            A[i] = Math.Min(B[i - 1], B[i]);
        }
        A[N - 1] = B[N - 2];

        Console.WriteLine(A.Sum());
    }
}

File: abc/abc140/abc140_c.fs
// https://atcoder.jp/contests/abc140/submissions/12019154

let N = stdin.ReadLine() |> int
let B = stdin.ReadLine().Split() |> Array.map int

[ 1..(N - 2) ]
|> List.map (fun i -> min B.[i - 1] B.[i])
|> List.sum
|> fun x -> B.[0] + x + B.[N - 2]
|> stdout.WriteLine
